### PR TITLE
[WIP] Parse ADNI XML metadata

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR continues the work started in #431 for adding metadata from XML files to the output of the ADNI to BIDS convertor. (Note: I made a lot of refactoring so I went for opening a new PR since the diff would be huge anyway...).

- [x] Refactor `adni_json.py`: shorter functions (especially `parse_xml_file`), more explicit variable and function names, docstrings, all functions private except  `create_json_metadata`...
- [x] Change output column names to be lower case (seems to better match [BIDS specs for tabular files](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#tabular-files)) 
- [ ] Add JSON file to better describe column names ?
- [ ] Add tests

### CODE REVIEW

**Comment by ghisvail** [comment]
Back when the first PR was proposed, I was wondering whether we could leverage the new [pandas.read_xml](https://pandas.pydata.org/docs/dev/reference/api/pandas.read_xml.html) to parse the XML metadata instead of raw `xml.etree` manipulations.

My thinking being: could we just let `pandas` do the actual XML parsing and only maintain code that selects and combines the metadata attributes we want? With the hypothesis that this would greatly simplify the resulting code and the future review of this PR.

**Comment by omar-rifai** [comment]
Thanks @NicolasGensollen  for picking this up. For the record, the two main elements that were blocking for merging [431](https://github.com/aramis-lab/clinica/pull/431) were (in order of priority):
 - Defining which metadata to include  (c.f discussion[#293](https://github.com/aramis-lab/clinica/discussions/293) and where they should be written in the BIDS hierarchy
 -  Eventually parallelizing with an appropriate library  (not a priority)

**Comment by omar-rifai** [comment]
@NicolasGensollen, before finalizing can you make sure to add @emaheux to the authors of the first commit as he provided the draft code for this. Thanks!

**Comment by NicolasGensollen** [comment]
@omar-rifai Of course ! I can also add you as the author of another one since I started from your work!

File: clinica/iotools/converters/adni_to_bids/adni_json.py [hunk 1]
```diff
@@ -284,6 +284,8 @@ def _get_image_metadata(img: xml.etree.ElementTree.Element) -> dict:
 
 def _get_image_rating(image_rating: xml.etree.ElementTree.Element) -> int:
```
Comment (ghisvail) [inline clinica/iotools/converters/adni_to_bids/adni_json.py:285]:
```suggestion
def _get_image_rating(image_rating: xml.etree.ElementTree.Element) -> Optional[int]:
```

File: test/tests/iotools/converters/adni_to_bids/test_adni_json.py [hunk 2]
```diff
@@ -108,37 +118,60 @@ def _load_xml_from_template(
 
 
 def _write_xml_example(base_path, template_id="ADNI_123_S_4567", **kwargs):
-    """TODO."""
-    xml_content = _load_xml_from_template(**kwargs)
-    xml_file_path = base_path / f"{template_id}.xml"
+    suffix = kwargs.pop("suffix", None)
```
Comment (ghisvail) [inline test/tests/iotools/converters/adni_to_bids/test_adni_json.py:121]: Better use `def _write_xml_example(..., suffix: Optional[str] = None, **kwargs)` and avoid `kwargs.pop`. Explicit is better than implicit.

Comment (NicolasGensollen) [reply test/tests/iotools/converters/adni_to_bids/test_adni_json.py]: I totally agree! Thanks! 👍

**Comment by omar-rifai** [comment]
Hello @NicolasGensollen, let us know if this is finalized so that we can review and merge. Thanks !

**Comment by NicolasGensollen** [comment]
Hi @omar-rifai 
I believe this is ready for reviews.
I re-ran the converter this afternoon (see folder `data_ci_2022_xml`, the output is in `ref`).

**Review by omar-rifai — APPROVED** [review]
Looks good thanks ! I'll merge after the rebase.

File: clinica/iotools/converters/adni_to_bids/adni_json.py [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+
+
+def _run_parsers(xml_files: list) -> Tuple[list, dict]:
+    """Run the parser `_parse_xml_file` on the list of files `xml_files`.
+    Returns a tuple consisting if parsed images
+    metadata and captured exceptions.
+
+    .. note::
+        Use multiprocessing / multithreading for parsing the files?
+        Not sure we will get a huge performance boost by doing that tbh.
```
[the remark this answers is not in the document]
Comment (NicolasGensollen) [reply clinica/iotools/converters/adni_to_bids/adni_json.py:446]: As discussed, we can remove the comment

**Comment by omar-rifai** [comment]
Probably related to the test itself but I'm getting an `TypeError` on `test_get_existing_scan_dataframe`:

```
 def test_get_existing_scan_dataframe(tmp_path):
        """Test function `_get_existing_scan_dataframe`."""
        from clinica.iotools.converters.adni_to_bids.adni_json import _get_existing_scan_dataframe
        from pandas.testing import assert_frame_equal
        subj_path = tmp_path / "sub-01"
        subj_path.mkdir()
>       with pytest.warns(match="No scan tsv file for subject sub-01 and session foo"):
E       TypeError: warns() missing 1 required positional argument: 'expected_warning'

```

**Comment by omar-rifai** [comment]
@NicolasGensollen, can you also please run a `make format` (again?) so that it passes the lint checks?

File: clinica/iotools/converters/adni_to_bids/adni_json.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (NicolasGensollen) [reply clinica/iotools/converters/adni_to_bids/adni_json.py]: Done in [d8b7d59](https://github.com/aramis-lab/clinica/pull/587/commits/d8b7d5965c7f9569623be3cabe7e48df9f30b7b5)

**Comment by NicolasGensollen** [comment]
>can you also please run a make format (again?)

Done in 6034162

> I'm getting an `TypeError` on `test_get_existing_scan_dataframe`

Oops, missed your comment on Friday.
Did you manage to run the tests since then?
Doing `pytest -v` works on my end.

**Comment by omar-rifai** [comment]
> > can you also please run a make format (again?)
> 
> Done in [6034162](https://github.com/aramis-lab/clinica/commit/60341625eadb8cb3b68b92a782751ac3358dc709)
> 
> > I'm getting an `TypeError` on `test_get_existing_scan_dataframe`
> 
> Oops, missed your comment on Friday. Did you manage to run the tests since then? Doing `pytest -v` works on my end.

Yes seems to work now, thanks! I also fixed some issues with the `data_ci` and the converters succeed now. However, It seems that there is still an issue with `merge_tsv` in iotools. Can you check that is everything merged ? I had a similar issue last week from a recent change in dev.

**Pull request merged**